receiver/awss3receiver/s3reader.go

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awss3receiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver"
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type s3Reader struct {
+	listObjectsClient ListObjectsAPI
+	getObjectClient   GetObjectAPI
+	s3Bucket          string
+	s3Prefix          string
+	filePrefix        string
+	startTime         time.Time
+	endTime           time.Time
+	timeStep          time.Duration
+	getTimeKey        func(time.Time) string
+}
+
+type s3ReaderDataCallback func(context.Context, string, []byte) error
+
+func newS3Reader(cfg *Config) (*s3Reader, error) {
+	listObjectsClient, getObjectClient, err := newS3Client(cfg.S3Downloader)
+	if err != nil {
+		return nil, err
+	}
+	startTime, err := parseTime(cfg.StartTime)
+	if err != nil {
+		return nil, err
+	}
+	endTime, err := parseTime(cfg.EndTime)
+	if err != nil {
+		return nil, err
+	}
+
+	var getTimeKey func(time.Time) string
+	var timeStep time.Duration
+	if cfg.S3Downloader.S3Partition == "hour" {
```

Comment:
Out of ignorance, why is this time granularity referred to as `S3 partition`? I don't have much context here, so I don't understand the relevance to partition.

Reply:
The reason for S3Partition name is due to this reflecting what is already being used by the S3 exporter. It makes sense to use the same config names when the receiver is the complement of the exporter.